lizard_connector/connector.py

```diff
@@ -208,18 +213,90 @@ def paginated(self):
         return bool(self.next_url)
 
 
-class Timeseries(Endpoint):
-    endpoint = 'timeseries'
-
-
-class Rasters(Endpoint):
-    endpoint = 'rasters'
-
-
-class Events(Endpoint):
-    endpoint = 'events'
+class Datatype(object):
+    """
+    Experimental class that forms the building block for data types.
 
+    Each data type contains a list of queries for each relevant lizard-api
+    endpoint. These are the only viable queries for each endpoint.
+    """
 
-# TODO:
-# class Assets(Endpoint):
-#     endpoint = ''
+    @property
+    def queries(self):
+        """
+        This property needs to be implemented for the class to function.
+        """
+        raise NotImplementedError
+
+    def list_endpoints(self):
+        """Lists all available endpoints for this datatype."""
+        return list(self.queries.keys())
+
+    def download(self, endpoint, *querydicts, **queries):
+        """Downloads a json as a dict from an endpoint."""
+        endpoint_ = Endpoint(endpoint)
+        return endpoint_.download(*querydicts, **queries)
+
+
+class Timeseries(Datatype):
+    queries = {
+        "timeseries": {
+            "in_bbox": functools.partial(
```

Comment:
Ik moest even googlen wat `functools.partial` ook al weer precies deed.

a) ook weer wat her-leerd

b) dit is niet gelijk voor iedereen leesbaar

c) het is een tikkie creatief: wijst het erop dat een iets andere organisatie/api/etc beter/duidelijker zou zijn? Ik weet het niet direct, maar dat is het soort vragen dat ik mezelf stel als ik zoiets zie.